src/pmod_gps.erl

```diff
@@ -40,36 +53,45 @@ start_link(Slot, _Opts) ->
 
 % @doc Get the GPS data.
 %
-% The input parameter specifies which sentence to get. For a description of
-% the sentences see the
+% The input parameter specifies which type of sentence to get.
+% For a description of the sentences see the
 % <a href="https://reference.digilentinc.com/_media/reference/pmod/pmodgps/pmodgps_rm.pdf">
 % PmodGPS Reference Manual
+% The sentence CRC is checked for all the sentence types, but for now only the
+% GGA values are parsed. If other sentences are needed, grisp_nmea needs to be
+% extended to support more types.
 % </a>.
 %
 % === Example ===
 % ```
 %  2> pmod_gps:get(gga).
-%  <<"$GPGGA,145832.000,5207.3597,N,01135.6957,E,1,5,2.50,61.9,M,46.7,M,,*6F\n">>
+%  {gps,gga,#{fixed => false,alt => undefined,lat => undefined,long => undefined,time => 26799}}
 %  3> pmod_gps:get(gsa).
-%  <<"$GPGSA,A,3,17,06,19,02,24,,,,,,,,2.69,2.51,0.97*0B\n">>
+%  {gps,gsa,<<"A,3,17,06,19,02,24,,,,,,,,2.69,2.51,0.97">>}
 %  4> pmod_gps:get(gsv).
-%  <<"$GPGSV,3,3,12,14,22,317,17,17,10,040,35,29,09,203,,22,02,351,*7F\n">>
+%  {gps,gsv,<<"3,3,12,14,22,317,17,17,10,040,35,29,09,203,,22,02,351">>}
 %  5> pmod_gps:get(rmc).
-%  <<"$GPRMC,150007.000,A,5207.3592,N,01135.6895,E,0.46,255.74,120220,,,A*64\n">>
+%  {gps,rmc,<<"150007.000,A,5207.3592,N,01135.6895,E,0.46,255.74,120220,,,A">>}
 %  6> pmod_gps:get(vtg).
-%  <<"$GPVTG,297.56,T,,M,0.65,N,1.21,K,A*33\n">>
+%  {gps,vtg,<<"297.56,T,,M,0.65,N,1.21,K,A">>}
 % '''
--spec get('gga' | 'gsa' | 'gsv' | 'rmc' | 'vtg') -> binary().
-get(Sentence) ->
-    call({get, Sentence}).
+-spec get(grisp_nmea:message_id()) ->
+    {grisp_nmea:talker_id(), grisp_nmea:message_id(), map() | binary()} | undefined.
+get(MessageId) ->
```

Comment:
Shouldn't we pattern match for the supported message IDs here?

Reply:
I was wondering ab out that too. Because the nmea parsing library returns binaries instead of atoms if the talker id or message type is unknown. it is tropically possible to have any binary for the message id. So what I do is fill the map with all known possible values and fail if there is no key, so this way if the parsing gives a custom message id it would still be possible to retrieve it.

Reply:
Alright, that sounds good.